src/queryResult/queryResultWebViewController.ts

```diff
@@ -309,4 +325,22 @@ export class QueryResultWebviewController extends ReactWebviewViewController<
         const messageText = messages.join("\n");
         await this._vscodeWrapper.clipboardWriteText(messageText);
     }
+
+    public getNumExecutionPlanResultSets(
+        resultSetSummaries: qr.QueryResultWebviewState["resultSetSummaries"],
+        actualPlanEnabled: boolean,
+    ): number {
+        const summariesLength = Object.keys(resultSetSummaries).length;
+        if (!actualPlanEnabled) {
+            return summariesLength;
+        }
+        // if actual plan is enabled, then
```

Comment:
Is this statement always true, like even for queries that have no results, like PRINTs or INSERTs, etc?  I don't know, if just seems like we may be "taking a leap of faith" here.

Reply:
Agreed.  Seems better to actually inspect each summary.  What does ADS do to differentiate between types?

Reply:
I changed it to manually check